### PR TITLE
Run st2_e2e_tests in parallel (besides robot tests)

### PR DESCRIPTION
**Pull request by humblearner**

### CODE REVIEW

File: actions/workflows/st2_e2e_tests.yaml [hunk 1]
```diff
@@ -138,105 +141,117 @@ workflows:
...
-            #     input:
-            #         hosts: <% $.host %>
-            #         env: <% $.env %>
-            #         cmd: st2 run tests.test_key_triggers <% $.st2_cli_args %>
-            #         timeout: 600
-            #     on-success:
-            #         - test_quickstart_trace
+                # on-success:
+                #     - test_quickstart_key_triggers
+            test_quickstart_key_triggers:
```
Comment (humblearner) [inline actions/workflows/st2_e2e_tests.yaml:237]: Reverting https://github.com/StackStorm/st2cd/pull/210 as the fix is there for keystore scope `st2kv.system.` https://github.com/StackStorm/st2tests/pull/67  and st2tests has branching as well: https://github.com/StackStorm/st2cd/pull/211

**Comment by arm4b** [comment]
@humblearner Awesome! :+1: 

Curious how that'd run in prod.

From what I see, we might have also some policies in place, limiting any parallel execution.
See: https://github.com/StackStorm/st2ci/tree/master/policies
Probably requires adjusting/removing those limits.

**Comment by humblearner** [comment]
@armab: Yes, besides that ^^, v2.3 didn't have changes from master in st2tests: https://github.com/StackStorm/st2tests/pull/109. And, disabling the policy doesn't help. Trying to fine tune it, a little more.

**Comment by Kami** [comment]
Just curious - do you have any timing info?

How does this speed things up? :)

**Comment by arm4b** [comment]
+1 same curiosity as @Kami 

Because previously we typically were afraid to run heavy things in parallel via mistral workflows. Eg. how that works in reality and what needs to be adjusted/what are consequences/stability.

**Comment by humblearner** [comment]
Before: 20+minutes
After: 10-12 minutes

**Comment by arm4b** [comment]
@humblearner Nice :+1: 

How does that specific improvement affected overall build timing? (from start to end). A the moment we have the following time to wait for build status: 
- Ubuntu16 e2e: `18mins` ([example](https://stackstorm.slack.com/archives/C02FUU8MT/p1502046018096624))
- RHEL7 e2e: `21mins` ([example](https://stackstorm.slack.com/archives/C02FUU8MT/p1502046017096443))

Any observations on this side?

**Comment by humblearner** [comment]
@armab : The examples you gave is the current time: 

- Ubuntu16 e2e: `18mins` ([example](https://stackstorm.slack.com/archives/C02FUU8MT/p1502046018096624))
- RHEL7 e2e: `21mins` ([example](https://stackstorm.slack.com/archives/C02FUU8MT/p1502046017096443))


Earlier it was:

- Ubuntu16 e2e: `27mins` ([example](https://stackstorm.slack.com/archives/C02FUU8MT/p1496258842961052))
- RHEL7 e2e: `28mins` ([example](https://stackstorm.slack.com/archives/C02FUU8MT/p1496086097353517))

**Comment by arm4b** [comment]
Nice! :+1:
Yes, the full build timing diff numbers is what I'm interested in.

**Pull request merged**